File: src/main/java/com/server/ordering/repository/FoodRepository.java
package com.server.ordering.repository;

import com.server.ordering.domain.Food;
import lombok.RequiredArgsConstructor;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceException;
import java.util.List;

@Repository
@RequiredArgsConstructor
public class FoodRepository {

    private final EntityManager em;
    private final JdbcTemplate jdbcTemplate;

    public void save(Food food) {
        em.persist(food);
    }

    public Food findOne(Long id) {
        return em.find(Food.class, id);
    }

    public List<Food> findAll(Long restaurantId) {
        return em.createQuery("select distinct m from Food m where m.restaurant.id =:id", Food.class)
                .setParameter("id", restaurantId)
                .getResultList();
    }

    public void remove(Food food) throws PersistenceException {
        em.remove(food);
    }

    public void remove(Long foodId) {
        jdbcTemplate.update("delete from food where food_id=?", foodId);
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/repository/FoodRepository.kt
package com.byeoru.ordering_server.repository

import com.byeoru.ordering_server.domain.Food
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Repository
import javax.persistence.EntityManager
import javax.persistence.PersistenceException

@Repository
class FoodRepository(private val em: EntityManager,
                     private val jdbcTemplate: JdbcTemplate) {

    fun save(food: Food) {
        em.persist(food)
    }

    fun findOne(id: Long): Food {
        return em.find(Food::class.java, id)
    }

    fun findAll(restaurantId: Long): List<Food> {
        return em.createQuery("select distinct m from Food m where m.restaurant.id =:id", Food::class.java)
            .setParameter("id", restaurantId)
            .resultList
    }

    @Throws(PersistenceException::class)
    fun remove(food: Food) {
        em.remove(food)
    }

    fun remove(foodId: Long) {
        jdbcTemplate.update("delete from food where food_id=?", foodId)
    }
}